sets `stripe_fullname` and `stripe_email` in the payment form card te…

jagui opened the pull request:
sets `stripe_fullname` and `stripe_email` in the payment form card template so that they can be used when confirming the card payment intent

clotaire202 commented:
Thank you for the PR.

Could you please tell us why you need the email ? Is it for tracking to display something to the customer ?
Do you think others data could be interesting ?

jagui commented:
Hi @clotaire202 this PR allows to easily cross reference orders between the PS backoffice and the Stripe dashboard. This email is displayed in the `Customer` column in the dashboard and the name when you drill down into the transaction details. Very helpful also for quickly spotting fraudulent payments with very dissimilar emails and usernames

clotaire202 commented:
Hi @jagui 

The customer details in the Stripe Dashboard is to manage the save card (and retrieve feature)
The next release (2.1.0, now in develop) will include this feature. That's why we have conflicts with you branch. A new table on the ps database will store the id customer and the customer stripe key to retrieve or update data.
I think we need to wait the next release.

Thank you.

jagui commented:
Fair enough. Do you have an ETA for version 2.1.0?
Cheers, 
Juan

clotaire202 commented:
Tests are OK, we wait for translations (EN and FR is OK). And we are looking for beta testers if you want.

jagui commented:
I can definitely help w/ ES translation and I might look into beta testing, though I'm pretty busy developing a private module

clotaire202 commented:
Hi @jagui 

2.1.0 is now released and published on PrestaShop addons. You can activate save card in order to save customer details on Stripe Dashboard.

jagui committed:
updates the customer's firstname, lastname, email and country's code on `hookPayment`

Rationale: these fields are only set on the module's `hookHeader`, and that works as long as there's a registered customer at that point. However on one-page-checkout scenarios, the customer might not be set at that point.
Typically, this customer will already be set once `hookPayment` is called, hence we can update the name, email and country variables on the main script, so that they're actually sent with the intent confirmation.

jagui commented:
T

> Hi @jagui
> 
> 2.1.0 is now released and published on PrestaShop addons. You can activate save card in order to save customer details on Stripe Dashboard.

HI again @clotaire202. I've already tested the 2.1.0 version, but I'm still experiencing the same problem. I'm sure it works fine with regular checkout flows, but with one-page-checkout the info about the customer on `hookHeader` is not set yet, and it needs to be updated once the customer has been created. I'm doing the update via the `payment_form_card.tpl`
On top of that, I've noticed that the payment intent confirmation is failing due to an invalid ISO 2-letter country code (which happens when the customer's address is not yet set and the value passed by `hookHeader` is `false`.

Hope you can reconsider this. Cheers, Juan

clotaire202 commented:
I'll check you proposal. https://github.com/202-ecommerce/stripe_official/blob/feature/2.2.0/_dev/js/payments.js

jagui commented:
Have you tried with the buit-in 1 page checkout flow? I use [this module](https://www.knowband.com/prestashop-plugins/prestashop-checkout-and-orders/prestashop-one-page-supercheckout), which btw will have to adapt to https://github.com/202-ecommerce/stripe_official/pull/3

Also please note that you're basically doing the same trick with the `stripe_amount` variable: it might have changed since the header was sent, therefore you update it.

Anyhow I'll take a look at the changes you sent. Thanks for the prompt response, I appreciate your time. 

Cheers, 
Juan

mribeiro202 commented:
Hello jagui,
Sorry for our late reply, we have updated the module several times since we last talked. Can you update the module with the latest version (2.3.1) and tell us if your problem persists.

clotairer commented:
This PR is now out of date. We might close this PR.

Many thanks for your contribution.
